crates/rpc/src/server/api.rs

```diff
@@ -392,6 +398,32 @@ impl api_server::Api for RpcService {
             ))
         })?;
 
+        // If transaction inputs are provided, re-execute the transaction to validate it.
+        if let Some(tx_inputs_bytes) = &request.transaction_inputs {
```

Comment:
Would it make sense to move the body of this if-statement into a separate helper function?